contracts/adapters/swap/astroport/src/contract.rs

```diff
@@ -158,20 +159,25 @@ fn execute_swap(
     // Create a response object to return
     let mut response: Response = Response::new().add_attribute("action", "execute_swap");
 
-    // Add an astroport pool swap message to the response for each swap operation
-    for operation in &operations {
-        let swap_msg = WasmMsg::Execute {
-            contract_addr: env.contract.address.to_string(),
-            msg: to_json_binary(&ExecuteMsg::AstroportPoolSwap {
-                operation: operation.clone(),
-            })?,
-            funds: vec![],
-        };
-        response = response.add_message(swap_msg);
+    for route in &routes {
+        // Add an astroport pool swap message to the response for each swap operation
+        for operation in &route.operations {
+            let swap_msg = WasmMsg::Execute {
```

Comment:
Without changing the ExecuteMsg::AstroportPoolSwap, this would not do as intended for a split route where a route is given and wants each trade to do have of the in asset in since the swap function queries the contract balance and swaps it all in